### PR TITLE
Amit feature/splash screen

### PR DESCRIPTION
**Pull request by amit-metri**
Splash screen implementation using SplashScreenAPI from Android 12

Below files are changed
       modified:   app/build.gradle > Upgraded compile and target SDK version 31 to include SplashScreenAPI 
       modified:   app/src/main/AndroidManifest.xml > Theme.Spalsh is applied to MainActivity
       modified:   app/src/main/java/ca/bc/gov/health/ircreader/MainActivity.kt > init block placed
       modified:   app/src/main/res/values/themes.xml > Theme.Splash is created.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -92,4 +92,6 @@ dependencies {
     androidTestImplementation "com.google.dagger:hilt-android-testing:$versions.hilt"
 
     kaptAndroidTest "com.google.dagger:hilt-android-compiler:$versions.hilt"
+
+    implementation "androidx.core:core-splashscreen:1.0.0-alpha01"
```
Comment (pinakinkansara) [inline app/build.gradle:96]: @amit-metri  Please declare version in versions.gradle & reference it here.

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -17,7 +17,8 @@
         <activity
             android:name=".MainActivity"
             android:exported="true"
-            android:screenOrientation="portrait">
+            android:screenOrientation="portrait"
+            android:theme="@style/Theme.MySplash">
```
Comment (pinakinkansara) [inline app/src/main/AndroidManifest.xml:21]: @amit-metri  rename "Theme.MySplash" to "Theme.BC_VAX.Splash

File: app/src/main/res/values/themes.xml [hunk 3]
```diff
@@ -1,4 +1,10 @@
 <resources xmlns:tools="http://schemas.android.com/tools">
+    <!-- Splash screen theme -->
+    <style name="Theme.MySplash" parent="Theme.SplashScreen">
```
Comment (pinakinkansara) [inline app/src/main/res/values/themes.xml:3]: rename theme Splash theme to Theme.BC_VAX.Splash

File: .idea/.name [hunk 4]
```diff
@@ -0,0 +1 @@
+BCVAX-Android
```
Comment (pinakinkansara) [inline .idea/.name:1]: add this file itself to .gitIgnore

File: app/build.gradle (hunk 1 above)
Comment (amit-metri) [reply app/build.gradle]: done

File: app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (amit-metri) [reply app/src/main/AndroidManifest.xml]: done

File: app/src/main/res/values/themes.xml (hunk 3 above)
Comment (amit-metri) [reply app/src/main/res/values/themes.xml]: done

File: .idea/.name (hunk 4 above)
Comment (amit-metri) [reply .idea/.name:1]: done

**Review by pinakinkansara — APPROVED** [review]

**Pull request merged**